### PR TITLE
Search service

### PR DESCRIPTION
**Pull request by Paramveer12**
Screenshot_20200606_115810
Observable need fixing

### CODE REVIEW

**Comment by wayne9499** [comment]
add screenshots

**Comment by lakshay-angrish** [comment]
add visual description aids a.k.a. screenshots

File: server.js [hunk 1]
```diff
@@ -140,6 +141,28 @@ app.put('/deleteTask', (req, res) => {
...
+app.post('/searchTask',(req,res) => {
+  var noMatch = null;
+
+    const regex = new RegExp(escapeRegex(req.body.text), 'gi');
+    Task.find({title: regex},function(err, data){
+      if(err){
+        console.log(err);
+      }else{
+        if(data.length < 1) {
+          noMatch = "No campgrounds match that query, please try again.";
```
Comment (lakshay-angrish) [inline server.js:153]: You know what to do here

File: src/app/tasks-layout/tasks-layout.component.ts [hunk 2]
```diff
@@ -3,39 +3,55 @@ import { HttpClient } from '@angular/common/http';
...
+    // if(this.searchResults.length == 0){
+
     this.http.get('http://localhost:3000/allTasks', { responseType: 'json' }).subscribe(
       (response: any[]) => {
         this.tasksToday = [];
         this.tasksUpcoming = [];
         this.tasksCompleted = [];
+         this.H1 = 'Due Today';
+         this.H2 = 'Upcoming';
+         this.H3 = 'Completed';
```
Comment (lakshay-angrish) [inline src/app/tasks-layout/tasks-layout.component.ts:54]: where's H4?

Comment (Paramveer12) [reply src/app/tasks-layout/tasks-layout.component.ts:54]: Search results

File: server.js (hunk 1 above)
Comment (Paramveer12) [reply server.js:153]: That's if there's no match, forgot to change the string

**Comment by Paramveer12** [comment]
They are supposed to refresh and show only the search results in tasks layout but there's something wrong with the refreshing part, everything is showing simultaneously

File: src/app/tasks-layout/tasks-layout.component.ts (hunk 2 above)
Comment (lakshay-angrish) [reply src/app/tasks-layout/tasks-layout.component.ts:54]: "Where" have you assigned H4 its value? I see the said assignment as commented code just below.

Comment (Paramveer12) [reply src/app/tasks-layout/tasks-layout.component.ts:54]: It's not working, that's why it's commented.Imma try to fix it today

File: src/app/tasks-layout/tasks-layout.component.ts [hunk 3]
```diff
@@ -3,39 +3,54 @@ import { HttpClient } from '@angular/common/http';
 import { MatDialog, MatDialogConfig } from '@angular/material/dialog';
 import { EditTaskComponent } from '../user/edit-task/edit-task.component';
 import { ReloadService } from '../reload.service';
+import {SearchService} from '../search.service';
+import { convertActionBinding } from '@angular/compiler/src/compiler_util/expression_converter';
```
Comment (lakshay-angrish) [inline src/app/tasks-layout/tasks-layout.component.ts:7]: If this is not being used, it should not be imported

File: src/app/header/header.component.ts [hunk 4]
```diff
@@ -20,5 +27,27 @@ export class HeaderComponent implements OnInit {
...
+        for (const task of response) {
+          this.searchData.push(task);
+        }
+        this.data.changeMessage(this.searchData)
+      },
+      (error) => {
+        alert(error);
+      }
+    );
+
```
Comment (lakshay-angrish) [inline src/app/header/header.component.ts:46]: remove unnecessary newlines

**Comment by lakshay-angrish** [comment]
Your date is showing incorrectly. Convert it to dateString()

**Pull request merged**